tests/implementations/register.test.ts

```diff
@@ -0,0 +1,60 @@
+import EnvRegistry from '../../src/EnvRegistry';
+import { environments } from './mocks';
+import { validationMessages } from '../../src/utils';
+
+describe('Register test suite', () => {
+  let envRegistry: EnvRegistry;
+
+  beforeEach(() => {
+    localStorage.clear();
+    envRegistry = new EnvRegistry('token');
+  });
+
+  it('Calling register method registers the version of the provided envKey and Env ', async (done) => {
+    expect.assertions(1);
+    await envRegistry.register({ envKey: 'master', env: environments.master});
+    envRegistry.environments$({}).subscribe((env) => {
+      expect(env).toEqual({key: 'master', value: environments.master });
+      done();
+    });
+  });
+
+  it('Calling register method with providing not valid envKey and Env', () => {
+    expect.assertions(16);
+
+    const badEnvKeyValues = [null, undefined, 123, [], ['test'], {}, { test: 'test' }];
+    badEnvKeyValues.forEach((envKey) => {
+      // @ts-ignore
+      envRegistry.register({ envKey, env: environments.develop })
+        .catch((err: any) => expect(err).toEqual(new Error(validationMessages.envKeyIsNotCorrect)));
+    });
+
+    const badEnvValues = [null, undefined, 123, 'test', [], ['test'], {}, { test: 'test' }, { accessPoints: [] }];
+    badEnvValues.forEach((env) => {
+      // @ts-ignore
+      envRegistry.register({ envKey: 'master', env })
+        .catch((err: any) => expect(err).toEqual(new Error(validationMessages.envIsNotCorrect)));
+    });
+  });
+
+  it('Calling register method with an envKey already registered processes to update', async (done) => {
+    expect.assertions(2);
+    await envRegistry.register({ envKey: 'test', env: environments.develop});
+    envRegistry.environments$({}).subscribe((env) => {
+      expect(env).toEqual({ key: 'test', value: { accessPoints: [
+            { url: 'http://accessPoint/dev/service1' },
+            { url: 'http://accessPoint/dev/service2' }
+          ]}});
+    });
+    await envRegistry.register({ envKey: 'test', env: environments.master });
+    envRegistry.environments$({}).subscribe((env) => {
+      expect(env).toEqual({ key: 'test', value: { accessPoints: [
```

Comment:
create a variable from the expected value to show that it's the same in both cases

Reply:
do you want to check that **createRepository** has been called both times?

Reply:
Only once but yes